tools/promql-generator/promql-generator.go

```diff
@@ -0,0 +1,354 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
Given this file is based on the example from https://github.com/cortexproject/promqlsmith, we should include a provenance comment here as well.